### PR TITLE
Here's a README.

### PR DESCRIPTION
**Pull request by phs**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,80 @@
...
+
+Chef itself does not offer a way to cleanup removed entries from a run list as a first class feature, perhaps because we can emulate this ourselves.  For example, one can imagine our administrator adding a `disable_service_daemon` "cleanup" recipe to the node at the same time that the original daemon recipe is removed.
+
+That works, but it requires that the clean up recipe exists and that the administrator is aware of the issue and so can take action.  This however scales poorly: consider that the offending daemons might be brought as transitive cookbook dependencies.  It would better if Chef could do this for us.
+
+It turns out that it can, with some help.  This cookbook supplies two LWRPs named `dishes_dish` and `dishes_sink`, which work together to enable Chef (and recipe authors) to react to recipes removed from a run list by including corrosponding "cleanup" recipes.
```
Comment (miroswan) [inline README.md]: Check spelling on corresponding

File: README.md [hunk 2]
```diff
@@ -1,2 +1,80 @@
...
+  sink 'default'
+end
+
+# Include the wash recipes of all dishes whose recipe is not in the run list
+dishes_sink 'default'
+```
+
+One attaches a cleanup (or "wash") recipe to a monitored recipe with `dishes_dish`.  The association is kept in a hidden normal attribute and thus lives in the state of the Chef server.  Nothing is saved on the node itself.  Each recipe that needs cleanup actions should use a dish.
+
+A recipe is allowed to use its _own_ dish, allowing cookbooks to clean up after themselves (but see below about sinks) however it is not required.  Dishes and corrosponding wash recipes can also live in a wrapper cookbook.
```
Comment (miroswan) [inline README.md]: This may need clarification.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,80 @@
...
+dishes_dish 'cookbook::daemon' do
+  action :use
+  wash_recipe 'cookbook::disable_daemon'
+  sink 'default'
+end
+
+# Include the wash recipes of all dishes whose recipe is not in the run list
+dishes_sink 'default'
+```
+
```
Comment (miroswan) [inline README.md:34]: Might want to explain what happens in this scenario. I'm assuming from the previous text that the wash_recipe will get to work when cookbook::daemon is removed from the run_list. We might want to be explicit about this here though.

File: README.md (hunk 1 above)
Comment (phs) [inline README.md]: Done

File: README.md (hunk 3 above)
Comment (phs) [inline README.md:34]: How about

> In this example, the `cookbook::daemon` recipe uses a dish.  The dish declares that when `cookbook::daemon` vanishes from the run list, the `dishes_sink` should respond by including the `cookbook::disable_daemon` recipe.

File: README.md (hunk 2 above)
Comment (phs) [inline README.md]: Fair enough, but I'm not sure which part needs it.  What should I clarify?

**Pull request merged**